src/containers/quiz/question-answer/Answer.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  FormControlLabel,
+  FormControlLabelProps,
+  Box,
+  Checkbox,
+  Radio,
+  InputAdornment,
+  SxProps
+} from '@mui/material'
+import { ChangeEventHandler, FC } from 'react'
+import CheckIcon from '@mui/icons-material/Check'
+import CloseIcon from '@mui/icons-material/Close'
+import { spliceSx } from '~/utils/helper-functions'
+import { styles } from './Answer.styles'
+import { QuestionTypesEnum } from '~/types'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import { questionType } from '~/components/question-editor/QuestionEditor.constants'
+import { useTranslation } from 'react-i18next'
+
+interface Answer {
+  text: string
+  isCorrect?: boolean
+  value?: string
+  label?: string
+  checked?: boolean
+  showCorrectness?: boolean
+  type: QuestionTypesEnum
+  isEditable?: boolean
+  sx?: SxProps
+  onTextInputChange?: ChangeEventHandler<HTMLInputElement>
+  onCheckboxChange?: (isChecked: boolean) => void
+}
+
+const Answer: FC<Answer> = ({
+  text,
+  isCorrect,
+  checked,
+  type,
+  label,
+  sx,
+  onTextInputChange,
+  onCheckboxChange,
+  value = '',
+  showCorrectness = false,
+  isEditable = true
+}) => {
+  const { isMultipleChoice, isOpenAnswer } = questionType(type)
+
+  const { t } = useTranslation()
+
+  const shouldShowState = (isOpenAnswer || checked) && showCorrectness
+  const baseState = isCorrect ? 'correct' : 'incorrect'
+  const answerState = shouldShowState ? baseState : undefined
+
+  const style = styles({ state: answerState, isOpenAnswer })
```

Comment:
I think it is not consistent with the rest of our codebase. Please consider using more modular styles, like `styles.root(isOpenAnswer, bgcolor)` or `styles.icon(color)`

Reply:
Thanks, fixed